boot/freeldr/freeldr/ntldr/arch/i386/winldr.c

```diff
@@ -525,15 +525,19 @@ WinLdrSetProcessorContext(void)
      * Longhorn/Vista reports LimitLow == 0x0fff == MM_PAGE_SIZE - 1, whereas
      * Windows 7+ uses larger sizes there (not aligned on a page boundary).
      */
-#if 1
-    /* Server 2003 way */
-    KiSetGdtEntryEx(KiGetGdtEntry(pGdt, KGDT_R0_PCR), (ULONG32)Pcr, 0x1,
-                    TYPE_DATA, DPL_SYSTEM, TRUE, 2);
-#else
-    /* Vista+ way */
-    KiSetGdtEntry(KiGetGdtEntry(pGdt, KGDT_R0_PCR), (ULONG32)Pcr, MM_PAGE_SIZE - 1,
-                  TYPE_DATA, DPL_SYSTEM, 2);
-#endif
+    if (OperatingSystemVersion < _WIN32_WINNT_VISTA)
+    {
+        /* Server 2003 way */
+        KiSetGdtEntryEx(KiGetGdtEntry(pGdt, KGDT_R0_PCR), (ULONG32)Pcr, 0x1,
+                        TYPE_DATA, DPL_SYSTEM, TRUE, 2);
+    }
+    else
+    {
+        /* Vista+ way */
```

Comment:
By the way @DarkFire01 , what's described here is from observable behaviour.
Do you think there could be a reason for this change (if so, why)?

Reply:
It's weird because it more so affects SMP iirc, which is bizarre to me. 
I'd have to verify this